system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -6180,15 +6114,20 @@ class CLdapClient : implements ILdapClient, public CInterface
         return addResource(RT_FILE_SCOPE, user, resource, PT_ADMINISTRATORS_AND_USER, m_ldapconfig->getResourceBasedn(RT_FILE_SCOPE));
     }
 
-    virtual aindex_t getManagedScopeTree(SecResourceType rtype, const char * basedn, IArrayOf<ISecResource>& scopes)
+    virtual aindex_t getManagedScopeTree(LDAP* ld, SecResourceType rtype, const char * basedn, IArrayOf<ISecResource>& scopes)
     {
+        Owned<ILdapConnection> lconn;
+        if (nullptr == ld)
+        {
+            lconn.set(m_connections->getConnection());
+            ld = ((CLdapConnection*)lconn.get())->getLd();
```

Comment:
Why is this cast here? This function being called is the only function in the interface.  Same for the other code above.